hack/local-up-cluster.sh

```diff
@@ -688,111 +687,113 @@ function start_kubelet {
+       cloud_config_arg+=("--provider-id=$(hostname)")
     fi
 
     mkdir -p "/var/lib/kubelet" &>/dev/null || sudo mkdir -p "/var/lib/kubelet"
     # Enable dns
     if [[ "${ENABLE_CLUSTER_DNS}" = true ]]; then
       if [[ "${ENABLE_NODELOCAL_DNS:-}" == "true" ]]; then
-        dns_args="--cluster-dns=${LOCAL_DNS_IP} --cluster-domain=${DNS_DOMAIN}"
+        dns_args=("--cluster-dns=${LOCAL_DNS_IP}" "--cluster-domain=${DNS_DOMAIN}")
       else
-        dns_args="--cluster-dns=${DNS_SERVER_IP} --cluster-domain=${DNS_DOMAIN}"
+        dns_args=("--cluster-dns=${DNS_SERVER_IP}" "--cluster-domain=${DNS_DOMAIN}")
       fi
     else
       # To start a private DNS server set ENABLE_CLUSTER_DNS and
       # DNS_SERVER_IP/DOMAIN. This will at least provide a working
       # DNS server for real world hostnames.
-      dns_args="--cluster-dns=8.8.8.8"
+      dns_args=("--cluster-dns=8.8.8.8")
     fi
-    net_plugin_args=""
+    net_plugin_args=()
     if [[ -n "${NET_PLUGIN}" ]]; then
-      net_plugin_args="--network-plugin=${NET_PLUGIN}"
+      net_plugin_args=("--network-plugin=${NET_PLUGIN}")
     fi
 
-    auth_args=""
+    auth_args=()
     if [[ "${KUBELET_AUTHORIZATION_WEBHOOK:-}" != "false" ]]; then
-      auth_args="${auth_args} --authorization-mode=Webhook"
+      auth_args+=("--authorization-mode=Webhook")
     fi
     if [[ "${KUBELET_AUTHENTICATION_WEBHOOK:-}" != "false" ]]; then
-      auth_args="${auth_args} --authentication-token-webhook"
+      auth_args+=("--authentication-token-webhook")
     fi
     if [[ -n "${CLIENT_CA_FILE:-}" ]]; then
-      auth_args="${auth_args} --client-ca-file=${CLIENT_CA_FILE}"
+      auth_args+=("--client-ca-file=${CLIENT_CA_FILE}")
     else
-      auth_args="${auth_args} --client-ca-file=${CERT_DIR}/client-ca.crt"
+      auth_args+=("--client-ca-file=${CERT_DIR}/client-ca.crt")
     fi
 
-    cni_conf_dir_args=""
+    cni_conf_dir_args=()
     if [[ -n "${CNI_CONF_DIR}" ]]; then
-      cni_conf_dir_args="--cni-conf-dir=${CNI_CONF_DIR}"
+      cni_conf_dir_args=("--cni-conf-dir=${CNI_CONF_DIR}")
     fi
 
-    cni_bin_dir_args=""
+    cni_bin_dir_args=()
     if [[ -n "${CNI_BIN_DIR}" ]]; then
-      cni_bin_dir_args="--cni-bin-dir=${CNI_BIN_DIR}"
+      cni_bin_dir_args=("--cni-bin-dir=${CNI_BIN_DIR}")
     fi
 
-    container_runtime_endpoint_args=""
+    container_runtime_endpoint_args=()
     if [[ -n "${CONTAINER_RUNTIME_ENDPOINT}" ]]; then
-      container_runtime_endpoint_args="--container-runtime-endpoint=${CONTAINER_RUNTIME_ENDPOINT}"
+      container_runtime_endpoint_args=("--container-runtime-endpoint=${CONTAINER_RUNTIME_ENDPOINT}")
     fi
 
-    image_service_endpoint_args=""
+    image_service_endpoint_args=()
     if [[ -n "${IMAGE_SERVICE_ENDPOINT}" ]]; then
-      image_service_endpoint_args="--image-service-endpoint=${IMAGE_SERVICE_ENDPOINT}"
+      image_service_endpoint_args=("--image-service-endpoint=${IMAGE_SERVICE_ENDPOINT}")
     fi
 
+    # shellcheck disable=SC2206
```

Comment:
I think it should not be changed to avoid behavior changes.
So I ignore this message.

In ./hack/local-up-cluster.sh line 781:
      ${KUBELET_FLAGS}
      ^--------------^ SC2206: Quote to prevent word splitting, or split robustly with mapfile or read -a.